bee-node/bee-node/src/tools/convert.rs

```diff
@@ -0,0 +1,136 @@
+        hrp: String,
+    },
+}
+
+pub fn exec(tool: &ConvertTool) -> Result<(), ConvertError> {
+    match tool {
+        ConvertTool::Bech32ToHex { bech32 } => {
+            let hex = bech32_to_hex(bech32.as_str());
+            match hex {
+                Ok(_) => println!("Your Hex encoded address is:\t{}", hex.unwrap()),
+                Err(e) => println!("Error: {}", e),
+            }
+        }
+        ConvertTool::HexToBech32 { hex, hrp } => {
+            let bech32 = hex_to_bech32(hex.as_str(), hrp.as_str());
+            match bech32 {
+                Ok(_) => println!("Your Bech32 address is:\t{:?}", bech32.unwrap()),
+                Err(e) => println!("Error: {}", e),
+            }
+        }
+        ConvertTool::HexPubkeyToBech32 { pubkey, hrp } => {
+            let bech32 = hex_public_key_to_bech32_address(pubkey.as_str(), hrp.as_str());
+            match bech32 {
+                Ok(_) => println!("Your Bech32 address is:\t{:?}", bech32.unwrap()),
+                Err(e) => println!("Error: {}", e),
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Transforms bech32 to hex
+fn bech32_to_hex(bech32: &str) -> Result<String, ConvertError> {
+    let address = Address::try_from_bech32(bech32).map_err(|_| ConvertError::InvalidAddress())?;
+    let hex_string = match address {
+        (_, Address::Ed25519(ed)) => ed.to_string(),
+        (_, Address::Alias(alias)) => alias.to_string(),
+        (_, Address::Nft(nft)) => nft.to_string(),
+    };
+    Ok(hex_string)
+}
+
+/// Transforms a hex encoded address to a bech32 encoded address
+fn hex_to_bech32(hex: &str, bech32_hrp: &str) -> Result<String, ConvertError> {
+    let address: Ed25519Address = hex
+        .parse::<Ed25519Address>()
+        .map_err(|_| ConvertError::InvalidAddress())?;
+    Ok(Address::Ed25519(address).to_bech32(bech32_hrp))
+}
+
+/// Transforms a hex encoded public key to a bech32 encoded address
+fn hex_public_key_to_bech32_address(hex: &str, bech32_hrp: &str) -> Result<String, ConvertError> {
+    let mut public_key = [0u8; Ed25519Address::LENGTH];
+    hex::decode_to_slice(&hex, &mut public_key).map_err(|_| ConvertError::InvalidAddressLength())?;
+
+    let address = Blake2b256::digest(&public_key)
+        .try_into()
+        .map_err(|_e| ConvertError::InvalidAddress())?;
+    let address: Ed25519Address = Ed25519Address::new(address);
+    Ok(Address::Ed25519(address).to_bech32(bech32_hrp))
+}
+
+#[cfg(test)]
+mod bech32tests {
+    use crate::tools::convert::*;
+    // spec: https://github.com/iotaledger/tips/blob/main/tips/TIP-0011/tip-0011.md
+    #[test]
+    fn bech32_to_hex() {
+        let bech32_to_hex = ConvertTool::Bech32ToHex {
+            bech32: "iota1qrhacyfwlcnzkvzteumekfkrrwks98mpdm37cj4xx3drvmjvnep6xqgyzyx".to_string(),
+        };
+        // output: "0xefdc112efe262b304bcf379b26c31bad029f616ee3ec4aa6345a366e4c9e43a3";
+        if let Err(e) = exec(&bech32_to_hex) {
+            println!("{:?}", e)
```

Comment:
The tests just print an error if there is one, but without looking manually at the output, it's not visible if it fails or not when running `cargo test`.  Using `.unwrap()` would make the test fail as it should if there is an error

What about having `exec(&bech32_to_hex).unwrap()` in the tests with `exec()` and also calling the three convert functions `bech32_to_hex()`, `hex_to_bech32()`,  `hex_public_key_to_bech32_address()` with an `assert_eq!(outputs, expected_output)` instead?

Something like this in each test (haven't tested if it compiles)
```Rust
fn bech32_to_hex() {
    let bech32_address = "iota1qrhacyfwlcnzkvzteumekfkrrwks98mpdm37cj4xx3drvmjvnep6xqgyzyx".to_string();

    let bech32_to_hex = ConvertTool::Bech32ToHex { bech32: bech32_address };

    exec(&bech32_to_hex).unwrap();

    let hex_encoded_address = bech32_to_hex(bech32_address).unwrap();

    assert_eq!(
        &hex_encoded_address,
        "0xefdc112efe262b304bcf379b26c31bad029f616ee3ec4aa6345a366e4c9e43a3"
    );
}
```